tools/terraform/fleetdm_client/fleetdm_client_test.go

```diff
@@ -0,0 +1,89 @@
+package fleetdm_client
+
+import (
+	"encoding/json"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"os"
+	"testing"
+)
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyz")
+var defaultDescription = "Awesome description"
+var defaultAgentOptions = `{"config":{"decorators":{"load":["SELECT uuid AS host_uuid FROM system_info;","SELECT hostname AS hostname FROM system_info;"]},"options":{"disable_distributed":false,"distributed_interval":10,"distributed_plugin":"tls","distributed_tls_max_attempts":3,"logger_tls_endpoint":"/api/osquery/log","logger_tls_period":10,"pack_delimiter":"/"}},"overrides":{}}`
+
+func randTeam() string {
+	b := make([]rune, 6)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return "aaa-" + string(b)
+}
+
+func TestBasic(t *testing.T) {
+	apiKey := os.Getenv("FLEETDM_APIKEY")
+	if apiKey == "" {
+		t.Skip("FLEETDM_APIKEY not set")
+	}
+	client := NewFleetDMClient("https://rbx.cloud.fleetdm.com", apiKey)
```

Comment:
This should probably hit a different URL for tests, but maybe that's something we can figure out post-merge.

Reply:
Oh yikes, didn't see this one.  Changed to take a FLEETDM_URL environment variable.